website/static/js/registrationUtils.js

```diff
@@ -646,9 +648,55 @@ Draft.prototype.submitForReview = function() {
             });
     }
     else {
-        self.beforeRegister(self.urls.submit.replace('{draft_pk}', self.pk));
+        return self.beforeRegister(self.urls.submit.replace('{draft_pk}', self.pk));
     }
 };
+Draft.prototype.approve = function() {
+    var ret = $.Deferred();
+    bootbox.dialog({
+        title: 'Before you continue...',
+        message: 'Are you sure you want to approve this submission? This action is irreversible',
+        buttons: {
+            cancel: {
+                label: 'Cancel',
+                className: 'btn-default',
+                callback: function() {
+                    bootbox.hideAll();
+                    ret.reject(); 
+                }
+            },
+            approve: {
+                label: 'Approve',
+                class: 'btn-danger',
+                callback: ret.resolve
+            }
+        }                
+    });
+    return ret.promise();
+};
+Draft.prototype.reject = function(confirmed) {
+    var ret = $.Deferred();
+    bootbox.dialog({
+        title: 'Before you continue...',
+        message: 'Are you sure you want to reject this submission? This action is irreversible',
+        buttons: {
+            cancel: {
+                label: 'Cancel',
+                className: 'btn-default',
+                callback: function() {
+                    bootbox.hideAll();
+                    ret.reject();
+                }
+            },
+            approve: {
+                label: 'Approve',
+                class: 'btn-danger',
+                callback: ret.resolve       
+            }
+        }                
+    });
+    return ret.promise();
```

Comment:
Couldn't the object passed to the `bootbox.dialog` be constructed once, send to approve and then cloned/updated when it's sent to the reject method? Cuts about 20 lines out of the file.